Data/Aeson/Types/FromJSON.hs

```diff
@@ -178,18 +178,42 @@ parseRealFloat _        Null       = pure (0/0)
 parseRealFloat expected v          = typeMismatch expected v
 {-# INLINE parseRealFloat #-}
 
+truncateIntegral :: Integral a => String -> Scientific -> Parser a
+truncateIntegral expected s
+    | Scientific.isInteger s = pure (truncate s)
```

Comment:
Here a similar comment applies as my previous comment. Instead of using both `isInteger` and `truncate` you can use the single [floatingOrInteger](http://hackage.haskell.org/package/scientific-0.3.4.8/docs/Data-Scientific.html#v:floatingOrInteger) which should be slightly more efficient and shorter.